libvncclient/rfbproto.c

```diff
@@ -1585,6 +1591,52 @@ SendKeyEvent(rfbClient* client, uint32_t key, rfbBool down)
   return WriteToRFBServer(client, (char *)&ke, sz_rfbKeyEventMsg);
 }
 
+/*
+ * Resize client
+ */
+rfbBool
+ResizeClientBuffer(rfbClient* client, int width, int height)
+{
+  client->width = width;
+  client->height = height;
+  client->updateRect.x = client->updateRect.y = 0;
+  client->updateRect.w = client->width;
+  client->updateRect.h = client->height;
+  return client->MallocFrameBuffer(client);
+}
+
+/*
+ * SendExtDesktopSize
+ */
+
+rfbBool
+SendExtDesktopSize(rfbClient* client, uint16_t width, uint16_t height)
+{
+  rfbSetDesktopSizeMsg sdm;
+  if (client->screen.width == 0 && client->screen.height == 0 ) {
+    rfbClientLog("Screen not yet received from server - not sending dimensions %dx%d\n", width, height);
+    return TRUE;
+  }
+  if (client->screen.width != rfbClientSwap16IfLE(width) && client->screen.height != rfbClientSwap16IfLE(height)) {
+    rfbClientLog("Sending dimensions %dx%d\n", width, height);
+    sdm.type = rfbSetDesktopSize;
+    sdm.width = rfbClientSwap16IfLE(width);
+    sdm.height = rfbClientSwap16IfLE(height);
+    sdm.numberOfScreens = 1;
+    client->screen.width = rfbClientSwap16IfLE(width);
+    client->screen.height = rfbClientSwap16IfLE(height);
+
+    if (!WriteToRFBServer(client, (char *)&sdm, sz_rfbSetDesktopSizeMsg))
+      return FALSE;
+    if (!WriteToRFBServer(client, (char *)&(client->screen), sz_rfbExtDesktopScreen))
+      return FALSE;
+
+    client->requestedResize = FALSE;
+
+    SendFramebufferUpdateRequest(client, 0, 0, 10, 10, FALSE);
```

Comment:
The values here seem bogus.

Reply:
Is this right?

`SendFramebufferUpdateRequest(client, 0, 0, width, height, FALSE);`